### DISCUSSION TITLE
RFC: LaTeX compiling standardisation

### DISCUSSION THREAD

**Discussion by Danysan1** [montali/unibo-ai#12] [Ideas]
I see lately lots of method are proliferating for compiling latex formulas into .md files. Sometimes these method clash against each other. IMHO we should decide together a common method, write it down in [CONTRIBUTING.md](CONTRIBUTING.md) and use only it fo future files.

Here's a list

Suppose this is the content of `foo.tex.md`:

```markdown
foo $bar^2$ quz
```

On GitHub the formula isn't translated:

> foo $bar^2$ quz

So until now many types of compilation have been used.

| Type | Tools | Compilation | Output | Example | View | [Dis]Advantages |
|-----|-----|-----|-----|-----|-----|-----|
| **Raw** | [VS Code extension "Markdown+Math"](https://marketplace.visualstudio.com/items?itemName=goessner.mdmath) |  | NO |  | Only as VS Code tab ([screenshot](https://user-images.githubusercontent.com/8406735/122568767-65dbe800-d04a-11eb-826c-4019c151c408.png)) | No compilation needed; useless for GitHub; works only on VS Code |
| **readme2tex** | [`readme2tex`](https://github.com/leegao/readme2tex) | See [CONTRIBUTING](https://github.com/montali/unibo-ai/blob/master/CONTRIBUTING.md#usage) | `./svgs/....svg` new; `foo.md` new ([example](https://raw.githubusercontent.com/Danysan1/unibo-ai/test/readme2tex/foo.md)); `foo.tex.md` unchanged | [Link](https://github.com/Danysan1/unibo-ai/blob/test/readme2tex/foo.md) | [Ok on GitHub web](https://user-images.githubusercontent.com/8406735/122572240-d7696580-d04d-11eb-89c4-7249f58c1b96.png),  [not ok on GitHub app](https://user-images.githubusercontent.com/8406735/122571439-059a7580-d04d-11eb-9d44-b70bbd1b4c5b.jpg) (maybe doesn't accept svg?) | Not limited to VS Code; leaves original unchanged (useful for subsequent changes); creates partial duplicates; often original remains behind compiled; requires python module manual installation |
| **Math>Image local** | [VS Code extension "Math to Image"](https://marketplace.visualstudio.com/items?itemName=MeowTeam.vscode-math-to-image) | [screenshot](https://user-images.githubusercontent.com/8406735/122571936-8b1e2580-d04d-11eb-88be-dd2ee7d1727c.png) | `/svg/....svg` new; `foo.tex.md` changed ([example](https://raw.githubusercontent.com/Danysan1/unibo-ai/test/math_image_local/foo.tex.md)) | [Link](https://github.com/Danysan1/unibo-ai/blob/test/math_image_local/foo.tex.md) | Ok on GitHub web, not ok on GitHub app | No duplicates; changes original (but leaving Latex formula in a comment); works only on VS Code |
| **Math>Image remote** | [VS Code extension "Math to Image"](https://marketplace.visualstudio.com/items?itemName=MeowTeam.vscode-math-to-image) | [screenshot](https://user-images.githubusercontent.com/8406735/122572100-b3a61f80-d04d-11eb-8b46-da8b9b6f55b9.png) | `foo.tex.md` changed ([example](https://raw.githubusercontent.com/Danysan1/unibo-ai/test/math_image_remote/foo.tex.md)) | [Link](https://github.com/Danysan1/unibo-ai/blob/test/math_image_remote/foo.tex.md) | Sometimes ok on GitHub web but sometimes no*, sometimes slower, not ok on GitHub app | No duplicates; changes original (but leaving Latex formula in a comment); with some hacking it can be done manually but mostly works only on VS Code |
| **tex-builder.py** | ? | ? | ? | ? | ? | ? |

(* sometimes a bad link is generated and the image is not rendered. For example when I tried inserting `$\left(\bigcup\limits_{c \ge 1} DTIME(n^c)\right)$` <img style="transform: translateY(0.1em); background: white;" src="https://raw.githubusercontent.com/Danysan1/unibo-ai/test/math_image_remote/svg/grl5K7IUo4.svg"> it generated the link [`https://render.githubusercontent.com/render/math?math=%5Cleft(%5Cbigcup%5Climits_%7Bc%20%5Cge%201%7D%20DTIME(n%5Ec)%5Cright)`](https://render.githubusercontent.com/render/math?math=%5Cleft(%5Cbigcup%5Climits_%7Bc%20%5Cge%201%7D%20DTIME(n%5Ec)%5Cright)) which does not work and caused the page to break.)

IMHO the best option is `Math>Image local` but I'm ok with the others

**Comment by montali** [comment]
You are absolutely right, and it is mostly my fault :(
I didn't think GitHub's LaTeX parser failed, so I'll agree with you on the `Image local` one!
